src/core/quantum_gate.jl

```diff
@@ -921,21 +969,21 @@ Return a gate that applies a rotation `theta` to the `target` qubit about the co
 
 The corresponding `Operator` is [`rotation(theta, phi)`](@ref).
 """
-rotation(target::Integer, theta::Real, phi::Real, T::Type{<:Complex}=ComplexF64) = Rotation(["R(θ=$(theta),ϕ=$(phi))"], "r", [target],
-    [theta, phi], T)
+rotation(target::Integer, theta::Real, phi::Real) = Rotation(target, [theta, phi])
 
-struct Rotation <: Gate
-    display_symbol::Vector{String}
-    instruction_symbol::String
-    target::SVector{1,Int}
+struct Rotation <: AbstractGate
+    target::Int
     parameters::SVector{2,Real}
```

Comment:
Rather name the parameters explicitly. It will be much clearer which one is theta and which one is phi.